docs/rfc/0003-importer-interface.md

```diff
@@ -0,0 +1,62 @@
+# Importer Interface RFC
+
+- Contribution Name: Importer Interface/Plugin Framework
+- Implementation Owner: Ganesh Vanahalli
+
+## Problem Statement
+
+Users would like to choose how to fetch blocks, either by downloading from an algod rest endpoint or directly from the gossip network or read blocks from a file which was previously written by the exporter.
+
+The current implementation of indexer only allows downloading blocks sequentially from an algod rest endpoint starting at the round number of recently written block to postgres db and users cannot choose a specific block to download.
+
+## Proposal
+
+Importer interface allows users to fetch any particular block either from algod rest endpoint (or directly from the network or from a file written to by the exporter plugin). Current interface also stores round number of the latest fetched block. 
+
+### Plugin Interface
+A sample interface is shown below:
+
+```GO
+package importerplugin
+
+// Importer defines the interface for importer plugin
+type Importer interface {
+	// Algod returns the importer's algod client
+	Algod() *algod.Client
+
+	// Round returns the recent downloaded block
+	Round() uint64
+
+	// Getblock takes the round number as an input and downloads that specific block, updates the 'lastRound' local variable and returns an encodedBlockCert struct consisting of Block and Certificate
+	GetBlock(rnd uint64) (rpcs.EncodedBlockCert, error)
+}
+```
+
+### Plugin implementation
+A sample implementation of importer:
+
+``` GO
+type importerImpl struct {
+	// aclient is the algod client used to fetch block
+	aclient   *algod.Client
+
+	// lastRound is the last successfully fetched block
+	lastRound uint64
+
+	// log.Logger given by the user through RegisterImporter method
+	log       *log.Logger
+
+	// To determine if the importer should fetch blocks from algod or gossip network or from a file
+	fetchMethod string
```

Comment:
It seems like this is meant to switch the import method between one of a few methods. Instead of defining this enum-like string and altering behavior of our implementation based on that, we should just have a separate `Importer` implementation for each fetching method and let the framework instantiate whichever one we want based on user input.

Reply:
Sounds good, I'll update it accordingly.